src/optimizer/remove_unused_columns.cpp

```diff
@@ -165,23 +166,6 @@ void RemoveUnusedColumns::VisitOperator(LogicalOperator &op) {
 			remove.VisitOperator(*child);
 		}
 		return;
-	case LogicalOperatorType::LOGICAL_ORDER_BY:
-		if (!everything_referenced) {
-			auto &order = op.Cast<LogicalOrder>();
-			D_ASSERT(order.projections.empty()); // should not yet be set
-			const auto all_bindings = order.GetColumnBindings();
-
-			for (idx_t col_idx = 0; col_idx < all_bindings.size(); col_idx++) {
-				if (column_references.find(all_bindings[col_idx]) != column_references.end()) {
-					order.projections.push_back(col_idx);
```

Comment:
I think this is removing an optimization that @lnkuiper added previously - is this intentional?

Reply:
We looked at this together, and managed to re-add the optimization, this time in the appropriate place, the `ColumnLifetimeAnalyzer`.